src/rust/ensogl/lib/list-view/src/entry.rs

```diff
@@ -114,6 +120,99 @@ impl ModelProvider for EmptyProvider {
 }
 
 
+// === Model Provider for Vectors ===
+
+/// An Entry Model Provider that wraps a Vec<T:Into<Model>>.
+#[derive(Clone,Debug)]
+pub struct VectorProvider<T> {
+    content : Vec<T>,
+}
+
+impl<T:Into<Model> + Debug + Clone> ModelProvider for VectorProvider<T> {
+    fn entry_count(&self) -> usize {
+        self.content.len()
+    }
+
+    fn get(&self, ix:usize) -> Option<Model> {
+        let item        = self.content.get(ix)?.clone();
+        let model:Model = item.into();
+        Some(model)
+    }
+}
+
+impl<T> From<Vec<T>> for VectorProvider<T> {
+    fn from(content: Vec<T>) -> Self {
+        VectorProvider{content}
+    }
+}
+
+
+// === Masked Model Provider ===
+
+/// An Entry Model Provider that wraps a Vec<T:Into<Model>> and allows the masking of a single item.
+#[derive(Clone,Debug)]
+pub struct SingleMaskedProvider {
+    content : AnyModelProvider,
+    mask    : Cell<Option<Id>>,
+}
+
+impl ModelProvider for SingleMaskedProvider {
+    fn entry_count(&self) -> usize {
+        match self.mask.get() {
+            None    => self.content.entry_count(),
+            Some(_) => self.content.entry_count().saturating_sub(1),
+        }
+    }
+
+    fn get(&self, ix:usize) -> Option<Model> {
+        let internal_ix = self.unmasked_index(ix);
+        self.content.get(internal_ix)
+    }
+}
+
+impl SingleMaskedProvider {
+
+    /// Return the index to the unmasked underlying data. Will only be valid to use after
+    /// calling `clear_mask`.
+    pub fn unmasked_index(&self, ix:Id) -> Id {
+        match self.mask.get() {
+            None                 => ix,
+            Some(id) if ix < id  => ix,
+            Some(_)              => ix+1,
+        }
+    }
```

Comment:
This needs better docs. After reading it I didnt getwhat really is "unmasked underlying data". It may be explained somewhere, but then thisdoc should point me to the explanation place.